### PR TITLE
Upgrade the publishing plugin

### PR DESCRIPTION
**Pull request by swankjesse**
Also rename KtBridgeGradlePlugin to ZiplinePlugin

Also move the KtBridge internals to app.cash.zipline.kotlinir

### CODE REVIEW

**Comment by JakeWharton** [comment]
I would prefer to stay on 0.14.2 until a version newer than 0.17.0 is released with a fix for the hostile env var check.

**Comment by swankjesse** [comment]
Is the hostile env var relevant if we get Github Actions to do all the publishing?

**Comment by JakeWharton** [comment]
Yes unfortunately it fails at configuration time meaning I can't do any builds because I have the "old" vars. Of course I need those for other projects which use an older version or alternative means of releasing to Sonatype.

**Comment by swankjesse** [comment]
Ooooh failing at configuration time is a nonstarter. Got it.

**Comment by JakeWharton** [comment]
Let me inquire for a new release in classic _user_ fashion.

**Comment by JakeWharton** [comment]
Asked

File: zipline-gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -21,11 +25,27 @@ buildConfig {
...
+    create("zipline") {
+      id = "app.cash.zipline"
+      displayName = "zipline"
+      description = "Compiler plugin to generate bridges between platforms"
+      implementationClass = "app.cash.zipline.gradle.ZiplinePlugin"
     }
   }
 }
+
+// The plugin marker artifacts don't conform to Maven Central's requirements. Disable publishing
```
Comment (JakeWharton) [inline zipline-gradle-plugin/build.gradle.kts]: This isn't a problem in licensee. Why is it here?

File: buildSrc/src/main/kotlin/Ext.kt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 object Ext {
   val compileSdkVersion = 29
   val ndkAbiFilters = listOf("x86", "x86_64", "armeabi-v7a", "arm64-v8a")
-  val kotlinPluginId = "app.cash.zipline.ktbridge.plugin"
+  val kotlinPluginId = "app.cash.zipline.kotlinir"
```
Comment (JakeWharton) [inline buildSrc/src/main/kotlin/Ext.kt]: Don't think using "ir" provides much value. In 6 months there will only be IR anyway. Can we do just .kotlin?

File: build.gradle.kts [hunk 3]
```diff
@@ -89,3 +84,50 @@ subprojects {
...
+        licenses {
+          license {
+            name.set("The Apache Software License, Version 2.0")
+            url.set("http://www.apache.org/licenses/LICENSE-2.0.txt")
+            distribution.set("repo")
+          }
+        }
+        developers {
+          developer {
+            id.set("square")
```
Comment (JakeWharton) [inline build.gradle.kts]:
```suggestion
            id.set("cashapp")
```

File: build.gradle.kts [hunk 4]
```diff
@@ -89,3 +84,50 @@ subprojects {
...
+            name.set("Square, Inc.")
```
Comment (JakeWharton) [inline build.gradle.kts]:
```suggestion
            name.set("Cash App")
```

**Review by JakeWharton — APPROVED** [review]
Approve based on the fact they said there'd be a new plugin version this week.

File: buildSrc/src/main/kotlin/Ext.kt (hunk 2 above)
Comment (swankjesse) [reply buildSrc/src/main/kotlin/Ext.kt]: I wanted `ir` to convey ASTs and code rewriting more than the IR API in particular. Maybe just saying `kotlin` felt unclear, and `kotlincompilerplugin` too verbose. I’ll go with `kotlin` unless you have other recommendations!

File: build.gradle.kts (hunk 3 above)
Comment (swankjesse) [reply build.gradle.kts]: I hope we’re not doing this .. . . so far as I know there’s no corporate entity for `cashapp` and we shouldn’t assign ownership to it?

File: build.gradle.kts (hunk 4 above)
Comment (swankjesse) [reply build.gradle.kts]: Same same.

File: zipline-gradle-plugin/build.gradle.kts (hunk 1 above)
Comment (swankjesse) [reply zipline-gradle-plugin/build.gradle.kts]: I get it this in Wire:

> Invalid POM: /com/squareup/wire/com.squareup.wire.gradle.plugin/4.0.0-alpha.12/com.squareup.wire.gradle.plugin-4.0.0-alpha.12.pom: Project name missing, Project description missing

Most likely cause is the publish plugin doesn’t create a complete POM for these.

Comment (swankjesse) [reply zipline-gradle-plugin/build.gradle.kts]: Found it!
https://github.com/square/wire/pull/2066/files

File: build.gradle.kts (hunk 3 above)
Comment (JakeWharton) [reply build.gradle.kts]: We do this on almost every cashapp project. There's no legal requirement here or anything. I think it ties it more closely to the GitHub org than anything.

Comment (swankjesse) [reply build.gradle.kts]: Works for me.

**Pull request merged**